From 487044b7b0f920b37743dc1df2fc87600c8661ba Mon Sep 17 00:00:00 2001
From: Lorenzo Colitti <lorenzo@google.com>
Date: Thu, 4 Jul 2013 00:12:40 +0900
Subject: [PATCH 2979/5965] net: ipv6: fix wrong ping_v6_sendmsg return value

ping_v6_sendmsg currently returns 0 on success. It should return
the number of bytes written instead.

Signed-off-by: Lorenzo Colitti <lorenzo@google.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 net/ipv6/ping.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/net/ipv6/ping.c b/net/ipv6/ping.c
index 9ca1f1ac7a14..858788da5a2f 100644
--- a/net/ipv6/ping.c
+++ b/net/ipv6/ping.c
@@ -214,5 +214,8 @@ int ping_v6_sendmsg(struct kiocb *iocb, struct sock *sk, struct msghdr *msg,
 	}
 	release_sock(sk);
 
-	return err;
+	if (err)
+		return err;
+
+	return len;
 }
-- 
2.19.0

